src/supervisor/watchers/handlers/index.ts

```diff
@@ -109,6 +109,15 @@ export function setupInformer(namespace: string, workloadKind: WorkloadKind) {
 
   const informer = makeInformer<KubernetesObject>(kubeConfig, namespacedEndpoint, loggedListMethod);
 
+  informer.on(ERROR, (err) => {
+    logger.error({ namespace, workloadKind, err }, 'informer error event occurred');
```

Comment:
Is it possible to distinguish "connection interrupted" because of, for example, AKS timeout vs. other errors? In the first case we can treat it as a warn (even info), and the latter as an actual error. We could then add a max retry attempt on the error case and end the program, as retrying is not always the best case (e.g. just spinning in an error loop that has no resolution).

Reply:
@ivanstanev I'm happy to narrow it down to `ECONNRESET`. And for other errors just log for now. WDYT?
```
{"name":"kubernetes-monitor","hostname":"snyk-monitor-779d797fc7-zddtk","pid":6,"level":50,"namespace":"snyk-monitor","workloadKind":"StatefulSet","err":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n    at TLSWrap.onStreamRead (internal/stream_base_commons.js:209:20)","code":"ECONNRESET"},"msg":"informer error event occurred","time":"2021-02-07T19:18:44.991Z","v":0}
```

Reply:
Sounds good! :)